app/assets/javascripts/components/common/text_input.jsx

```diff
@@ -1,116 +1,138 @@
-import React from 'react';
-import createReactClass from 'create-react-class';
+import React, { useRef } from 'react';
 import PropTypes from 'prop-types';
 import InputHOC from '../high_order/input_hoc.jsx';
 import Conditional from '../high_order/conditional.jsx';
 
-const TextInput = createReactClass({
-  displayName: 'TextInput',
+const TextInput = ({
+  value,
+  value_key,
+  editable,
+  label,
+  placeholder,
+  spacer,
+  valueClass,
+  p_tag_classname,
+  inline,
+  type,
+  max,
+  maxLength,
+  focus,
+  onBlur,
+  onClick,
+  append,
+  onKeyDown,
+  _value,
+  invalid,
+  onChange,
+  onFocus,
+  id,
+  children
+}) => {
+  const inputRef = useRef(null);
 
-  propTypes: {
-    value: PropTypes.any,
-    value_key: PropTypes.string,
-    editable: PropTypes.bool,
-    label: PropTypes.string,
-    placeholder: PropTypes.string,
-    spacer: PropTypes.string,
-    valueClass: PropTypes.string,
-    p_tag_classname: PropTypes.string,
-    inline: PropTypes.bool,
-    type: PropTypes.string,
-    max: PropTypes.string,
-    maxLength: PropTypes.string,
-    focus: PropTypes.func,
-    onBlur: PropTypes.func,
-    onClick: PropTypes.func,
-    append: PropTypes.node,
-    onKeyDown: PropTypes.func,
-    _value: PropTypes.any,
-    // validation: Regex used by Conditional
-    // required: bool used by Conditional
-  },
+  const onKeyDownHandler = (e) => {
+    if (!onKeyDown) return;
+    onKeyDown(e.keyCode, inputRef.current);
+  };
 
-  onKeyDown(e) {
-    if (!this.props.onKeyDown) { return; }
-    this.props.onKeyDown(e.keyCode, this.refs.inputbox);
-  },
+  let labelContent;
+  const usedSpacer = spacer || ': ';
 
-  dateChange(date) {
-    const value = date ? date.format('YYYY-MM-DD') : '';
-    return this.props.onChange({ target: { value } });
-  },
+  if (label) {
+    labelContent = label + usedSpacer;
+  }
```

Comment:
Can be simplified:

```
const labelContent = label ? `${label}${spacer || ': '}` : undefined;
```

Reply:
Done, I didn't touch those because I was under the impression to avoid ternary operators especially in react because they make the code more unreadable.